### ISSUE TITLE
assemble broken for Spring Boot and devtools since 2.3.0

### ISSUE THREAD

**Issue by jeffmaury** [fabric8io-images/s2i#171]
The s2i assemble script is supposed to explode the Spring Boot fat JAR if Spring Boot devtools dependency is found. It is broken since 2.3.0 as unzip is not found anymore in the image.

**Comment by vorburger** [comment]
@jeffmaury are you having the problem with the `centos` or the `rhel` variant of the base image?

**Comment by vorburger** [comment]
https://github.com/fabric8io-images/s2i/tree/master/java/images/centos#spring-boot-automatic-restarts

**Comment by vorburger** [comment]
@jeffmaury (or @dhirajsb or anyone else familiar with this stuff .. and FYI @edewit) I've toyed with this a bit and come up with #173 **_BUT_** (a) I cannot yet reproduce anything broken due to a missing `unzip`; (b) I don't really fully understand yet how it's supposed to work anyway... 

I could debug it further by looking at what #81 did in `assemble`, but it feels like I'm missing part of the picture here... perhaps yould could comment on [my example's README](https://github.com/vorburger/s2i/blob/spring-devtools/java/examples/spring-devtools/README.md) how to get that to actually work (and reproduce your problem) ?

**Comment by rhuss** [comment]
Not really remember how it works, just that the Spring Boot Devtools, which are required for hot-reload support, don't work on fat jar but only on the exploded one. 'would be interesting whether for new Spring Boot versions this is still the case.

**Comment by jeffmaury** [comment]
So the assemble script works that way:

1. checks if the fat jar contains spring-devtools
2. if true unzip the fat jar
3. then parse the MANIFEST.MF to get the Spring Boot application class name
4. then launch Java with specific parameters to use either the Spring Boot JAR ou WAR launcher and giving the Spring Boot application class name as argument

so step 2 is broken as it uses unzip which is not anymore in the image. Why don't we use jar for that ?

**Comment by rhuss** [comment]
> so step 2 is broken as it uses unzip which is not anymore in the image. Why don't we use jar for that ?

good point, should have been catched in the review as the feature has been introduced.

**Comment by vorburger** [comment]
> So the assemble script works that way:

@jeffmaury but why does [my example](https://github.com/vorburger/s2i/blob/spring-devtools/java/examples/spring-devtools/), where the fat fat JAR does contain spring-devtools, not cause the unzip and reproduce this problem? A reproducer (example) would be a useful first step here IMHO.

**Comment by jeffmaury** [comment]
I supposed your sample is correct but the script is wrong as unzip is not there and you need to check the logs to check for the exploded message

**Comment by vorburger** [comment]
> so step 2 is broken as it uses unzip which is not anymore in the image. Why don't we use jar for that ?

That's an idea, and we could do this e.g. in https://github.com/OASIS-learn-study/swissarmyknife-minecraft-server/pull/14/files, but here in this project it's not just [search & replace of "unzip" by "jar xvf"](https://github.com/fabric8io-images/s2i/search?q=unzip&unscoped_q=unzip), because of args (`-l` & `-p`)  - and I wouldn't want to raise a "blind" PR, without being able to test it (but see above; help with a reproducer test would be welcome!). 

Alternatively, therefore, and IMHO safer and better for anyone else this may affect, we should just fix the _regression_ we introduced in the move `FROM jboss/base-jdk:8` to `centos` were we lost `unzip` ? I'm proposing to do this in #185.

**Comment by vorburger** [comment]
@jeffmaury with the just merged #185 `unzip` is back in this image. 

Are you able to test to confirm if this issue is fixed? Using `master` OK, or do you need a release to Docker Hub? Do you just need a release of the upstream CentOS based community image, or also of downstream RHEL based one?

**Comment by m86194** [comment]
Using Spring Boot 2.1.3.RELEASE the code works, but the test is invalid as my fat jar does not include the dev tools jar.   Skipping that test and just look for the main class would in my opinion be sufficient.

**Comment by vorburger** [comment]
@m86194 a Pull Request with (more) fixes for this, or a working test, would probably be welcome. (FYI I'm no longer involved with this project, and just un-watched this issue.)
